libraries/rush-lib/src/utilities/Utilities.ts

```diff
@@ -773,39 +785,52 @@ export class Utilities {
 
     const escapedArgs: string[] = args.map((x) => Utilities.escapeShellParameter(x));
 
-    let result: child_process.SpawnSyncReturns<string | Buffer> = child_process.spawnSync(
-      escapedCommand,
-      escapedArgs,
-      options
-    );
+    const process: child_process.ChildProcess = child_process.spawn(escapedCommand, escapedArgs, options);
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    if (result.error && (result.error as any).errno === 'ENOENT') {
-      // This is a workaround for GitHub issue #25330
-      // https://github.com/nodejs/node-v0.x-archive/issues/25330
-      //
-      // TODO: The fully worked out solution for this problem is now provided by the "Executable" API
-      // from @rushstack/node-core-library
-      result = child_process.spawnSync(command + '.cmd', args, options);
+    const stdoutBuffers: string[] = [];
+    process.stdout?.on('data', (chunk) => {
+      stdoutBuffers.push(chunk.toString());
+    });
+
+    const stderrBuffers: string[] = [];
+    process.stderr?.on('data', (chunk) => {
+      stderrBuffers.push(chunk.toString());
+    });
+
+    let error: Error | undefined;
+    try {
+      await once(process, 'close');
+    } catch (e) {
+      error = e;
     }
 
-    Utilities._processResult(result);
-    return result;
+    const stderr: string = stderrBuffers.join('');
+    const stdout: string = stdoutBuffers.join('');
+
+    this._processResult({ error, stderr, status: process.exitCode });
+
+    return {
+      stdout,
+      stderr
+    };
   }
 
-  private static _processResult(result: child_process.SpawnSyncReturns<string | Buffer>): void {
-    if (result.error) {
-      result.error.message += '\n' + (result.stderr ? result.stderr.toString() + '\n' : '');
-      throw result.error;
+  private static _processResult({
+    error,
+    stderr,
+    status
+  }: {
+    error: Error | undefined;
+    stderr: string;
+    status: number | null;
+  }): void {
+    if (error) {
+      error.message += `\n${stderr}`;
+      throw error;
     }
 
-    if (result.status) {
-      throw new Error(
-        'The command failed with exit code ' +
-          result.status +
-          '\n' +
-          (result.stderr ? result.stderr.toString() : '')
-      );
+    if (status) {
+      throw new Error(`The command failed with exit code ${status}\n${stderr}`);
```

Comment:
Did non-zero exit codes always throw `spawnSync`? I know some tools intentionally return non-zero exit codes. Whatever we do here should be compatible with what `spawnSync` does.

Reply:
No, `spawnSync` returns an object that includes the exit code.